components/shared/profile/edit/EditSocials.tsx

```diff
@@ -0,0 +1,245 @@
+"use client";
+import React from "react";
+import {
+  Dialog,
+  DialogClose,
+  DialogContent,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@/components/ui/dialog";
+import { Input } from "@/components/ui/input";
+import CustomButton from "../../CustomButton";
+import { IUser } from "@/database/models/user.model";
+import twitter from "@/public/icons/icn-twitter.svg";
+import instagram from "@/public/icons/icn-instagram.svg";
+import linkedin from "@/public/icons/icn-linkedin.svg";
+import github from "@/public/icons/icn-github.svg";
+import dribble from "@/public/icons/icn-dribbble.svg";
+import facebook from "@/public/icons/icn-facebook.svg";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { X } from "lucide-react";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { SocialsSchema } from "@/lib/profileValidations";
+import Image from "next/image";
+import { Button } from "@/components/ui/button";
+
+import { updateUserSocials } from "@/lib/actions/user.actions";
+import { z } from "zod";
+
+interface EditSocialsProps {
+  user: Partial<IUser>;
+}
+
+const EditSocials = ({ user }: EditSocialsProps) => {
+  const { handleSubmit, register, setValue, getValues } = useForm({
+    resolver: zodResolver(SocialsSchema),
+    defaultValues: {
+      twitter: {
+        username: user.socials?.twitter?.username || "",
+        url: user.socials?.twitter?.url || "",
+      },
+      instagram: {
+        username: user.socials?.instagram?.username || "",
+        url: user.socials?.instagram?.url || "",
+      },
+      linkedin: {
+        username: user.socials?.linkedin?.username || "",
+        url: user.socials?.linkedin?.url || "",
+      },
+      github: {
+        username: user.socials?.github?.username || "",
+        url: user.socials?.github?.url || "",
+      },
+      dribbble: {
+        username: user.socials?.dribbble?.username || "",
+        url: user.socials?.dribbble?.url || "",
+      },
+      facebook: {
+        username: user.socials?.facebook?.username || "",
+        url: user.socials?.facebook?.url || "",
+      },
+    },
+  });
+  const onSubmit: SubmitHandler<z.infer<typeof SocialsSchema>> = async () => {
+    try {
+      const { twitter, instagram, linkedin, github, dribbble, facebook } =
+        getValues();
```

Comment:
```suggestion
  const onSubmit: SubmitHandler<z.infer<typeof SocialsSchema>> = async (values) => {
    try {
      const { twitter, instagram, linkedin, github, dribbble, facebook } =
        values
```

Values come as part of our submit